packages/sync/src/connect-indexdb.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { IndexeddbPersistence } from 'y-indexeddb';
+
+/**
+ * Internal dependencies
+ */
+import type { ObjectType, ObjectID, CRDTDoc } from './types';
+
+/**
+ * Connect function to the IndexedDB persistence provider.
+ *
+ * @param objectId   The object ID.
+ * @param objectType The object type.
+ * @param doc        The CRDT document.
+ *
+ * @return Promise that resolves when the connection is established.
+ */
+export function connectIndexDb(
+	objectId: ObjectID,
+	objectType: ObjectType,
+	doc: CRDTDoc
+): Promise< () => void > {
+	const roomName = `${ objectType }-${ objectId }`;
+	const provider = new IndexeddbPersistence( roomName, doc );
+
+	return new Promise( ( resolve ) => {
```

Comment:
Let us set the type on the `Promise` via the generic argument to make the Promise constructor type-safe and then let TS infer the type automatically

```suggestion
) {
	const roomName = `${ objectType }-${ objectId }`;
	const provider = new IndexeddbPersistence( roomName, doc );

	return new Promise< VoidFunction >( ( resolve ) => {
```